src/main/webapp/app/components/tokenInputGroups/TokenExtendButton.tsx

```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+import {
+  InputGroup,
+  Button,
+  Dropdown,
+  DropdownButton,
+  Row
+} from 'react-bootstrap';
+import classnames from 'classnames';
+import { getMomentInstance } from 'app/shared/utils/Utils';
+import DayPickerInput from 'react-day-picker/DayPickerInput';
+import 'react-day-picker/lib/style.css';
+import moment from 'moment';
+import {
+  APP_LOCAL_DATETIME_FORMAT_Z_FORCE,
+  APP_LOCAL_DATE_FORMAT
+} from 'app/config/constants';
+import MomentLocaleUtils from 'react-day-picker/moment';
+import { DefaultTooltip } from 'cbioportal-frontend-commons';
+
+type TokenExtendButtonProps = {
+  currentDate: string;
+  afterChangeDate: (newDate: string) => void;
+};
+
+export const TokenExtendButton: React.FunctionComponent<TokenExtendButtonProps> = props => {
+  let selectedDay: Date;
+
+  function handleDayChange(day: any) {
+    selectedDay = day;
+  }
+
+  function handleSelectedDay() {
+    props.afterChangeDate(
+      moment(selectedDay).format(APP_LOCAL_DATETIME_FORMAT_Z_FORCE)
+    );
+  }
+
+  const extendSchedule = [
+    {
+      amount: 1,
+      scheduleIndex: 1,
```

Comment:
I don't see the use case of adding this property. Can we just use the array index at the line 94?